api/python/t4/packages.py

```diff
@@ -158,18 +159,69 @@ def _clone(self):
 class Package(object):
     """ In-memory representation of a package """
 
-    def __init__(self, data=None, meta=None):
-        """
-        _data is of the form {logical_key: PackageEntry}
+    def __init__(self, name=None, pkg_hash=None, registry=None):
         """
-        self._data = data or {}
-        self._meta = meta or {}
+        Create a Package from scratch, or load one from a registry.
+
+        Args:
+            name(string): name of package to load
+            pkg_hash(string): top hash of package version to load
+            registry(string): location of registry to load package from
+        """
+        if name is None and pkg_hash is None and registry is None:
+            self._data = {}
+            self._meta = {'version': 'v0'}
+            return
+
+        if registry is None:
+            # default to local registry
+            registry = get_package_registry_root()
+
+        if pkg_hash is not None:
+            # if hash is specified, name doesn't matter
+            pkg_path = registry + 'packages/{}'.format(pkg_hash)
```

Comment:
maybe `'{}packages/{}'.format(registry, pkg_hash)` so you aren't mixing concat and format?